src/sourcekit-lsp/LanguageClientManager.ts

```diff
@@ -534,6 +540,20 @@ export class LanguageClientManager {
                         diagnostics
                     );
                 },
+                handleWorkDoneProgress: (() => {
+                    let restoringLSPStarted = false;
+                    return async (token, params, next) => {
+                        const result = await next(token, params);
+                        const title = (params as langclient.WorkDoneProgressBegin).title;
+                        if (title === "SourceKit-LSP: Restoring functionality") {
+                            restoringLSPStarted = true;
+                        } else if (params.kind === "end" && restoringLSPStarted) {
+                            restoringLSPStarted = false;
+                            await promptForDiagnostics(this.workspaceContext);
+                        }
+                        return result;
+                    };
```

Comment:
With https://github.com/apple/sourcekit-lsp/pull/1493 you can check whether the token has the `sourcekitd-crashed` prefix. 

Also, two things I just noticed:
1. I think this triggers the prompt for diagnostics whenever *any* work done progress ends after the `Restoring functionality` has started
2. We can show the prompt for diagnostics when sourcekitd crashes. We don’t need to wait for the service to be restored.

Reply:
I think there would be value in resetting the prompt status after 1 hour or maybe 1 day. It’s quite likely that if sourcekitd crashes again after a long time period, that it’s a different kind of issue that would be worth reporting.

Reply:
Sounds good, I've capped it to show the dialog at most once an hour.